src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,107 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskType;
+import duke.task.Todo;
+
+public class Parser {
+    /**
+     * Parses the input string coming from command-line.
+     * If there are any incomplete or invalid inputs, an exception is thrown.
+     *
+     * @param input The input string from the command-line.
+     * @return Command The command describing the action.
+     * @throws DukeException if the input syntax is invalid.
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        String[] tokens = input.split(" ", 2);
+        String[] args;
+        String[] times;
+        switch (tokens[0]) {

```

Comment:
This block of code is extremely neat and easily readable and understood. I have just noticed that writing switch-case statements in this way is much neater than writing deeply nested if-else statements that some us are used to.  And the way you abstract out all the try-catch statements into checkExistence has also helped in ensuring the code quality here. Well done! 👍